src/main/java/prefs/Preferences.java

```diff
@@ -26,7 +26,7 @@
     public GlobalConfig global;
 
     private Preferences(String configFileName, boolean createUnconditionally) {
-        this.fileHandler = new ConfigFileHandler(DIRECTORY, configFileName);
+        this.fileHandler = new ConfigFileHandler(DIRECTORY, configFileName, new GlobalConfig());
```

Comment:
Since we are going to make everything general, let's rename the variables to reflect this. It shall be `globalConfFileHandler` (or `globalCfh`, although I prefer the former since cfh is not very clear nor common) and `globalConfigFileName`.